src/BeneficiariesFile.hs

```diff
@@ -1,43 +1,68 @@
 module BeneficiariesFile (readBeneficiariesFile, Beneficiary) where
 
 import Config (Config (..))
-import Control.Monad ((<=<))
+import Control.Applicative ((<|>))
 import Data.Aeson.Extras (tryDecode)
-import Data.Either.Combinators (mapLeft, maybeToRight)
+import Data.Attoparsec.Text qualified as Attoparsec
+import Data.Either.Combinators (mapLeft, maybeToRight, rightToMaybe)
 import Data.Text (Text, lines, words)
 import Data.Text qualified as Text
 import Data.Text.IO (readFile)
 import FakePAB.Address (PubKeyAddress, deserialiseAddress, toPubKeyAddress)
+import FakePAB.UtxoParser qualified as UtxoParser
 import Ledger qualified
 import Ledger.Crypto (PubKeyHash (..))
+import Ledger.Value (AssetClass)
 import PlutusTx.Builtins (toBuiltin)
-import Text.Read (readEither)
+import Safe (atMay)
+import Text.Read (readMaybe)
 import Prelude hiding (lines, readFile, words)
 
 data Beneficiary = Beneficiary
-  { amount :: !Integer
-  , address :: !PubKeyAddress
+  { address :: !PubKeyAddress
+  , amount :: !Integer
+  , assetClass :: !AssetClass
   }
   deriving stock (Show)
 
 parseContent :: Config -> Text -> Either Text [Beneficiary]
-parseContent conf = traverse (parseBeneficiary conf) . lines
+parseContent conf =
+  mapLeft ("Beneficiaries file parse error: " <>)
+    . traverse (parseBeneficiary conf)
+    . lines
 
 parseBeneficiary :: Config -> Text -> Either Text Beneficiary
 parseBeneficiary conf = toBeneficiary . words
   where
     toBeneficiary :: [Text] -> Either Text Beneficiary
-    toBeneficiary [addr, amt] =
+    toBeneficiary strs =
       Beneficiary
-        <$> mapLeft (const "Invalid amount") (readEither (Text.unpack amt))
-        <*> parseAddress conf.usePubKeys addr
-    toBeneficiary _ = Left "Invalid format"
+        <$> parseAddress conf.usePubKeys (strs `atMay` 0)
+        <*> maybeToRight "Invalid amount" (parseAmt conf.dropAmount (strs `atMay` 1))
+        <*> maybeToRight "Invalid asset class" (parseAsset conf.assetClass (strs `atMay` 2))
```

Comment:
By the looks of this logic, if you don't put the amount it'll try to parse the asset class as the amount?
You may need to check the length of strs to get this logic right, or change the str indexes based on the config